show-device-location-using-indoor-positioning/src/main/java/com/esri/arcgismaps/sample/showdevicelocationusingindoorpositioning/MainActivity.kt

```diff
@@ -0,0 +1,361 @@
+            null
+        }
+    }
+
+    /**
+     * Retrieves the Levels tables which is an [ArcGISFeatureTable] that contains floor levels in accordance with the ArcGIS Indoors Information Model.
+     * Providing this table enables the retrieval of a location's floor level ID.
+     */
+    private fun getLevelsTable(): ArcGISFeatureTable? {
+        return try {
+            val levelsFeatureLayer =
+                mapView.map?.operationalLayers?.firstOrNull { it.name == "levels" } as? FeatureLayer
+            levelsFeatureLayer?.featureTable as? ArcGISFeatureTable
+        } catch (e: Exception) {
+            // if levels table not found in map's operationalLayers
+            showError("Levels table not found")
+            null
+        }
+    }
+
+    /**
+     * Sets up the location listeners, the navigation mode, display's the devices location as a blue dot, collect data source location changes and handles its status changes
+     */
+    private fun startLocationDisplay() {
+        val locationDisplay = mapView.locationDisplay.apply {
+            setAutoPanMode(LocationDisplayAutoPanMode.Navigation)
+            dataSource = mIndoorsLocationDataSource
+                ?: return showError("Error setting the IndoorsLocationDataSource value.")
+        }
+
+        // coroutine scope to start the location display, which will in-turn start IndoorsLocationDataSource to start receiving IPS updates.
+        lifecycleScope.launch {
+            locationDisplay.dataSource.start()
+        }
+
+        // coroutine scope to collect data source location changes like currentFloor, positionSource, transmitterCount, networkCount and horizontalAccuracy
+        lifecycleScope.launch {
+            locationDisplay.dataSource.locationChanged.collect { location ->
+                // get the location properties of the LocationDataSource
+                val locationProperties = location.additionalSourceProperties
+                if (locationProperties == null) {
+                    showError("LocationDataSource does not have any property-fields")
+                }
+                // retrieve information about the location of the device
+                val floor = (locationProperties["floor"] ?: "").toString()
+                val positionSource = (locationProperties["positionSource"] ?: "").toString()
+                val transmitterCount = (locationProperties["transmitterCount"] ?: "").toString()
+                val satelliteCount = (locationProperties["satelliteCount"] ?: "").toString()
+
+                // check if current floor hasn't been set or if the floor has changed
+                val newFloor = floor.toInt()
+                if (currentFloor == null || currentFloor != newFloor) {
+                    currentFloor = newFloor
+                    // update layer's definition express with the current floor
+                    val layerList = mapView.map?.operationalLayers
+                    layerList?.forEach { layer ->
```

Comment:
```suggestion
                    mapView.map?.operationalLayers?.forEach { layer ->
```
This could be simplified since we're not using the `layerList` variable.